src/controllers/main/main.ts

```diff
@@ -719,28 +741,30 @@ export class MainController extends EventEmitter {
       const learnedNewNfts = await this.portfolio.learnNfts(nfts, network.id)
       // update the portfolio only if new tokens were found through tracing
       if (learnedNewTokens || learnedNewNfts) {
-        this.portfolio
-          .updateSelectedAccount(
-            accountOp.accountAddr,
-            network,
-            getAccountOpsForSimulation(
-              account,
-              this.actions.visibleActionsQueue,
-              network,
-              accountOp
-            ),
-            { forceUpdate: true }
-          )
-          // fire an update request to refresh the warnings if any
-          .then(() => this.signAccountOp?.update({}))
+        await this.portfolio.updateSelectedAccount(
+          accountOp.accountAddr,
+          network,
+          getAccountOpsForSimulation(account, this.actions.visibleActionsQueue, network, accountOp),
+          { forceUpdate: true }
+        )
       }
+
+      if (this.signAccountOp)
+        this.signAccountOp.traceCallDiscoveryStatus = TraceCallDiscoveryStatus.Done
     } catch (e: any) {
+      if (this.signAccountOp)
+        this.signAccountOp.traceCallDiscoveryStatus = TraceCallDiscoveryStatus.Failed
+
       this.emitError({
         level: 'silent',
         message: 'Error in main.traceCall',
         error: e
       })
     }
+
+    this.signAccountOp?.calculateWarnings()
+    this.#traceCallTimeoutId = null
+    clearTimeout(timeoutId)
   }
```

Comment:
You've deleted
```
// fire an update request to refresh the warnings if any
 .then(() => this.signAccountOp?.update({}))
```
Please make sure that replacing it with `this.signAccountOp?.calculateWarnings()` is enough and there isn't some edge case. @borislav-itskov should know better

Reply:
The `calculateWarnings` call has been moved inside `traceCall` and now we're awaiting `updateSelectedAccount`. That leads be to believe we're covered here 👍